amp-beta-tester.php

```diff
@@ -0,0 +1,292 @@
+		}
+	}
+
+	return false;
+}
+
+/**
+ * Retrieves the current AMP update manifest, and updates it to include the analogous information
+ * from its GitHub release.
+ *
+ * @param object $release GitHub release JSON object.
+ * @return array|false Updated manifest, or false if it fails to retrieve the current update manifest.
+ */
+function generate_amp_update_manifest($release ) {
+	$current_manifest = get_current_amp_update_manifest();
+
+	if ( ! $current_manifest ) {
+		return false;
+	}
+
+	$manifest = [];
+	$zip_url  = get_download_url_from_amp_release( $release );
+
+	if ( $zip_url ) {
+		$manifest['package'] = $zip_url;
+	}
+
+	$manifest['new_version'] = $release->tag_name;
+	$manifest['url']         = $release->html_url;
+
+	return array_merge( (array) $current_manifest, $manifest );
+}
+
+/**
+ * Get the AMP plugin update manifest for the specified version from GitHub.
+ *
+ * @param string $version Version to get manifest for. Defaults to getting the latest pre-release.
+ * @return object|false Latest release, or false on failure.
+ */
+function get_amp_update_manifest($version = 'pre-release' ) {
+	$amp_manifest = null;
+	$releases     = get_site_transient( AMP__BETA__TESTER__RELEASES__TRANSIENT );
+
+	if ( empty( $releases ) ) {
+		$releases = get_amp_github_releases();
+		set_site_transient( AMP__BETA__TESTER__RELEASES__TRANSIENT, $releases, DAY_IN_SECONDS );
+	}
+
+	if ( is_array( $releases ) ) {
+		$amp_version = get_amp_version();
+
+		foreach ( $releases as $release ) {
+			if (
+				'pre-release' === $version
+				&& true === $release->prerelease
+				&& version_compare( $release->tag_name, $amp_version, '>' )
+			) {
+				$amp_manifest = generate_amp_update_manifest( $release );
+				break;
+			}
+
+			if ( $version === $release->tag_name ) {
+				$amp_manifest = generate_amp_update_manifest( $release );
+				break;
+			}
+		}
+	} else {
+		// Something went wrong fetching the releases.
+		return false;
+	}
+
+	if ( empty( $amp_manifest ) ) {
+		return false;
+	}
+
+	return (object) $amp_manifest;
+}
+
+/**
+ * Get the current AMP plugin update manifest.
+ *
+ * @return array|false Update manifest for current AMP plugin. False if it can't be retrieved.
+ */
+function get_current_amp_update_manifest() {
+	$updates = get_site_transient( 'update_plugins' );
+
+	if ( ! isset( $updates->response, $updates->no_update ) ) {
+		return false;
+	}
+
+	if ( isset( $updates->response[ AMP__PLUGIN__BASENAME ] ) ) {
+		$manifest = $updates->response[ AMP__PLUGIN__BASENAME ];
+	} elseif ( isset( $updates->no_update[ AMP__PLUGIN__BASENAME ] ) ) {
+		$manifest = $updates->no_update[ AMP__PLUGIN__BASENAME ];
+	} else {
+		return false;
+	}
+
+	return $manifest;
+}
+
+/**
+ * Determine if the supplied version code is a prerelease.
+ *
+ * @param string $plugin_version Plugin version code.
+ * @return bool
+ */
+function is_pre_release( $plugin_version ) {
+	return (bool) preg_match( '/^\d+\.\d+(\.\d+)?-/', $plugin_version );
```

Comment:
Maybe add an explanatory comment here about what the regex is checking.
```suggestion
	// A plugin version is a pre-release if it has a `-<suffix>`, so we're
	// checking for a hyphen after the version numbers.
	return (bool) preg_match( '/^\d+\.\d+(\.\d+)?-/', $plugin_version );
```

Reply:
No longer necessary as function has been removed.